presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -443,10 +443,10 @@ public ConnectorTableHandle getTableHandleForStatisticsCollection(ConnectorSessi
                     Iterable<List<Object>> records = () ->
                             stream(partitionManager.getPartitions(metastore, new HiveIdentity(session), sourceTableHandle, targetConstraint).getPartitions())
                                     .map(hivePartition ->
-                                            (List<Object>) IntStream.range(0, partitionColumns.size())
+                                            IntStream.range(0, partitionColumns.size())
                                                     .mapToObj(fieldIdToColumnHandle::get)
                                                     .map(columnHandle -> hivePartition.getKeys().get(columnHandle).getValue())
-                                                    .collect(toImmutableList()))
+                                                    .collect(toList()))
```

Comment:
add comment

and i suggest better exc msg (like `Fix NPE when reading from null partition`)